src/gui/main_window.cpp

```diff
@@ -232,6 +232,24 @@ void MainWindow::setIgnoreTouch(bool on)
 	ignore_touch_input = on;	
 }
 
+bool MainWindow::setIgnoreTouchAndWarn(bool on)
+{
+	if (on && !ignoreTouch())
+	{
+		QMessageBox dialog(QMessageBox::Warning, tr("Warning"),
+		                          tr("Display touch input will be ignored in the editor."),
+		                          QMessageBox::Cancel | QMessageBox::Ok);
```

Comment:
For ensuring stylus availability, maybe we just need an event filter on the OK button, sending warnings on touch-down, mouse-down, or key-down events, and otherwise dropping these events?

Reply:
I'm looking into it. However, I'm struggling with setting up the input event filtering on the `QPushButton` instance. For some reason, my code works on Linux but on Android, the OK button accepts the touch event. The current failing version of `setIgnoreTouchAndWarn()` is below. Still investigating.

```
bool MainWindow::setIgnoreTouchAndWarn(bool on)
{
        if (!on || ignoreTouch())
        {
                setIgnoreTouch(on);
                return true;
        }

        QMessageBox dialog(QMessageBox::Warning, tr("Warning"),
                           tr("The application will ignore display touch input beyond this point. Confirm that your alternate pointing device works by activating the Ok button."),
                           QMessageBox::Cancel | QMessageBox::Ok);
        dialog.setInformativeText(tr("If that does not work, go back to the editor settings and deactivate the \"Ignore display touch\" option."));
        dialog.setAttribute(Qt::WA_AcceptTouchEvents);
        dialog.raise();
        auto* ok_button = dialog.button(QMessageBox::Ok);
        ok_button->setAttribute(Qt::WA_AcceptTouchEvents);
        ok_button->installEventFilter(this);
        setIgnoreTouch(true);
        auto confirmed = dialog.exec() == QMessageBox::Ok;
        setIgnoreTouch(confirmed);

        return confirmed;
}
```

Reply:
Does the event filter consume (accept) the touch events, or do they fall through as unhandled, being resent as mouse event or similar?

Reply:
FTR, this code works as intended. It's based on plain `QDialog` and performs a complicated breakdance to reuse the event filter code. But it's at least a starting point.

Interesting observations:
1) The OK button receives focus even when it ignores the touch. The focus change is likely handled by the parent then.
2) Setting a filter on the parent dialog makes also the Back button inert to touch. It seems that the GUI event **filtering** works in the parent->child direction while the **event handling** is child->parent as hinted by `QObject::event()` docs. There is little explanation for these concepts or I've searched the wrong places.

```
bool MainWindow::setIgnoreTouchAndWarn(bool on)
{
	if (!on || ignoreTouch())
	{
		setIgnoreTouch(on);
		return true;
	}

	QDialog message(this);
	auto* layout = new QVBoxLayout();
	layout->addWidget(new QLabel(tr("Warning")));
	auto* text = new QLabel(tr("The application will ignore display touch input beyond this point. Confirm that your alternate pointing device works by activating the Ok button."));
	text->setWordWrap(true);
	layout->addWidget(text);
	auto* info = new QLabel(tr("If that does not work, go back to the editor settings and deactivate the \"Ignore display touch\" option."));
	info->setWordWrap(true);
	layout->addWidget(info);
	auto* button_layout = new QHBoxLayout();
	auto* button_back = new QPushButton(tr("Back"));
	button_layout->addWidget(button_back);
	button_layout->addStretch();
	auto* button_ok = new QPushButton(tr("OK"));
	button_layout->addWidget(button_ok);
	layout->addLayout(button_layout);
	message.setLayout(layout);
	connect(button_back, &QPushButton::clicked, &message, &QDialog::reject);
	qApp->removeEventFilter(this); // FIXME add activateEventFilter/deactivateEventFilter to MainWindow
	setIgnoreTouch(true);
	button_ok->setAttribute(Qt::WA_AcceptTouchEvents);
	button_ok->installEventFilter(this);
	connect(button_ok, &QPushButton::clicked, &message, &QDialog::accept);
	message.setWindowModality(Qt::WindowModal);
	auto confirmed = message.exec() == QDialog::Accepted;
	qApp->installEventFilter(this);
	QApplication::processEvents(QEventLoop::ExcludeUserInputEvents, 100 /* ms */);
	setIgnoreTouch(confirmed);
	
	return confirmed;
}
```